File: admin-web/src/api/customer.ts
import request from './request';

export interface CustomerVO {
  id: number;
  name: string;
  phone: string;
  address: string;
  tier: string;
  outstandingBalance: number;
  activeTickets: number;
  lifetimeOrders: number;
  totalSpent: number;
  status: number;
  createdAt: string;
}

export const customerApi = {
  list(params?: { page?: number; size?: number; keyword?: string; tier?: string }) {
    return request.get('/customers', { params });
  },
  get(id: number) {
    return request.get(`/customers/${id}`);
  },
  create(data: { name: string; phone: string; address: string; tier?: string; companyName?: string }) {
    return request.post('/customers', data);
  },
  update(id: number, data: Partial<CustomerVO>) {
    return request.put(`/customers/${id}`, data);
  },
  delete(id: number) {
    return request.delete(`/customers/${id}`);
  },
  recordPayment(id: number, amount: number) {
    return request.post(`/customers/${id}/payment`, { amount });
  },
};

File: backend/src/main/java/com/waterstation/ai/entity/Customer.java
package com.waterstation.ai.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@TableName("customer")
public class Customer {
    @TableId(type = IdType.AUTO)
    private Long id;
    private String name;
    private String phone;
    private String address;
    private String addressDetail;
    private BigDecimal latitude;
    private BigDecimal longitude;
    private String tier;
    private String companyName;
    private BigDecimal outstandingBalance;
    private Integer activeTickets;
    private Integer lifetimeOrders;
    private BigDecimal totalSpent;
    private String remark;
    private Integer status;
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
}
